src/main/java/com/github/swrirobotics/bags/BagService.java

```diff
@@ -580,6 +583,58 @@ public String getVehicleName(BagFile bag) {
         return null;
     }
 
+    public String getMetadata(BagFile bagFile) {
+        String[] topics = myConfigService.getConfiguration().getMetadataTopics();
+        try {
+            for (String topic : topics) {
+                com.github.swrirobotics.bags.reader.messages.serialization.MessageType mt = bagFile.getFirstMessageOnTopic(topic);
+                int index = 1;
+                try {
+                    // Advance to newest
+                    index++;
+                    mt = bagFile.getMessageOnTopicAtIndex(topic, index);
```

Comment:
For what it's worth, I think that using `forMessagesOnTopic` would probably be better than iterating over them with `getMessageOnTopicAtIndex` here.  It's a bit faster and you don't have to worry about catching an `IndexOutOfBoundsException`.